libraries/extractor/src/main/java/androidx/media3/extractor/DtsUtil.java

```diff
@@ -223,8 +357,262 @@ public static int getDtsFrameSize(byte[] data) {
+        0);
+  }
+
+  /**
+   * Returns the size of frame header in a DTS-HD frame by parsing a few bytes of data
+   * (minimum 8 bytes) from the input bitstream(containing only Extension Sub-stream).
+   *
+   * @param frame The initial 8 bytes(minimum) of a DTS-HD frame(having only Extension Sub-stream).
+   * @return Size of the DTS-HD frame header in bytes.
+   */
+  public static int parseDtsHdHeaderSize(byte[] frame) {
+    ParsableBitArray frameBits = getNormalizedFrameHeader(frame);
+    int nuBits4Header;
+    frameBits.skipBits( 32 + 8 + 2); // SYNC, UserDefinedBits, nExtSSIndex
+    // Unpack the num of bits to be used to read header size
+    int headerSizeType = frameBits.readBits(1);
+    if (headerSizeType == 0){
+      nuBits4Header = 8;
+    }
+    else{
+      nuBits4Header = 12;
+    }
+    // Unpack the substream header size
+    return frameBits.readBits(nuBits4Header) + 1;
+  }
+
+  /**
+   * Returns the DTS format given {@code data} containing the DTS-UHD(Profile 2) frame according to
+   * ETSI TS 103 491 V1.2.1 (2019-05), Section 6.4.3.
+   *
+   * @param frame The DTS-UHD frame to parse.
+   * @return The DTS format parsed from data in the header.
+   */
+  public static DtsFormatInfo parseDtsUhdFormat(byte[] frame) throws ParserException {
+    ParsableBitArray frameBits = getNormalizedFrameHeader(frame);
+    int sampleRate = 0;
+    int sampleCount = 0;
+    boolean bSyncFrameFlag = false;
+
+    int nSyncWord = frameBits.readBits(32);
+    if (nSyncWord == SYNC_VALUE_UHD_FTOC_SYNC_BE) {
+      bSyncFrameFlag = true;
+    }
+    int[] ucTable1 = new int[] { 5, 8, 10, 12 };
+    int nFTOCPayloadinBytes = extractVarLenBitFields(frameBits, ucTable1, true) + 1;
+
+    if (bSyncFrameFlag) {
+      // fullChannelBasedMixFlag, ETSI TS 103 491 V1.2.1, Section 6.4.6.1
+      if(frameBits.readBits(1) != 1) {
+        throw ParserException.createForMalformedContainer(
+            /* message= */ "Only supports full channel mask-based audio presentation",
+            /* cause= */ null);
+      }
+
+      int nBaseDurationIndex = frameBits.readBits(2);
+      sampleCount = BASE_DURATION[nBaseDurationIndex] * (frameBits.readBits(3) + 1);
+      int clockRateIndex = frameBits.readBits(2);
+      int clockRateHertz = 0;
+      switch (clockRateIndex) {
+        case 0:
+          clockRateHertz = 32000;
+          break;
+        case 1:
+          clockRateHertz = 44100;
+          break;
+        case 2:
+          clockRateHertz = 48000;
+          break;
+      }
+      // Read time stamp information
+      if (frameBits.readBits(1) == 1) {
+        frameBits.skipBits(32 + 4);
+      }
+      int samplRateMultiplier = (1 << frameBits.readBits(2));
+      sampleRate = clockRateHertz * samplRateMultiplier;
+    }
+
+    int chunkPayloadBytes = 0;
+    int numOfMDChunks = bSyncFrameFlag ? 1 : 0; // Metadata chunks
+    for (int nmdc = 0; nmdc < numOfMDChunks; nmdc++) {
+      int[] ucTable2 = new int[] { 6, 9, 12, 15 };
```

Comment:
AFAICT from the naming conventions in the spec, this is kinda like declaring int integer? Since uc just means unsigned char. Can you please define a better name for these tables?

Reply:
Yes, I will update this.

Reply:
This is done in the commit [ef41526](https://github.com/androidx/media/pull/275/commits/ef415268e215719ce385f5fb0e3149679a3bd91e)